plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/LocalizationPlugin.java

```diff
@@ -0,0 +1,114 @@
+package com.mapbox.mapboxsdk.plugins.testapp.activity;
+
+import android.support.annotation.NonNull;
+
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.style.layers.Layer;
+import com.mapbox.mapboxsdk.style.layers.PropertyFactory;
+
+import java.util.List;
+import java.util.Locale;
+
+/**
+ * The localization plugin enables localization of map labels into the user’s preferred language
+ * <p>
+ * Initialise this plugin in the {@link com.mapbox.mapboxsdk.maps.OnMapReadyCallback#onMapReady(MapboxMap)} and provide
+ * a valid instance of a {@link MapboxMap}.
+ * </p>
+ */
+public final class LocalizationPlugin {
+
+  private List<Layer> listOfMapLayers;
+
+  /**
+   * Create a localization plugin.
+   *
+   * @param mapboxMap the MapboxMap to apply localization plugin with
+   * @since 0.1.0
+   */
+  public LocalizationPlugin(@NonNull final MapboxMap mapboxMap) {
+    listOfMapLayers = mapboxMap.getLayers();
+    String deviceLanguage = Locale.getDefault().getLanguage();
+    for (Layer layer : listOfMapLayers) {
+      // TODO: Fix if() statement? What should if statement be looking for in layer ids?
+      if (layer.getId().contains("")) {
```

Comment:
Briefly looking over this, I think you are looking to change all Symbol layers? For that you can do `if(layer instanceof SymbolLayer) {}`

Reply:
Fixed